pkgs/_macro_tool/lib/cfe_macro_runner.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:frontend_server/compute_kernel.dart';
+import 'package:macro_service/macro_service.dart';
+import 'package:path/path.dart' as p;
+
+import 'macro_runner.dart';
+import 'source_file.dart';
+
+class CfeMacroRunner implements MacroRunner {
+  final String workspacePath;
+  final String packageConfigPath;
+
+  @override
+  final List<SourceFile> sourceFiles;
+
+  CfeMacroImplementation? cfeMacroImplementation;
+
+  CfeMacroRunner({required this.workspacePath, required this.packageConfigPath})
+      : sourceFiles = Directory(workspacePath)
+            .listSync(recursive: true)
+            .whereType<File>()
+            .where((f) => f.path.endsWith('.dart'))
+            .map((f) => SourceFile(f.path))
+            .toList() {}
+
+  void notifyChange(SourceFile sourceFile) {
+    throw UnimplementedError(
+        'CfeMacroRunner does not support incremental compilation.');
+  }
+
+  File get _productPlatformDill {
+    // TODO(davidmorgan): this dill comes from the Dart SDK running the test,
+    // but `package:frontend_server` and `package:front_end` are used as a
+    // library, so we will see version skew breakage. Find a better way.
+    final result = File(p.canonicalize('${Platform.resolvedExecutable}/../../'
+        'lib/_internal/vm_platform_strong_product.dill'));
+    if (!result.existsSync()) {
+      throw StateError('Failed to find platform dill: $result');
+    }
+    return result;
+  }
+
+  @override
+  Future<WorkspaceResult> run({bool injectImplementation = true}) async {
+    if (injectImplementation) {
+      cfeMacroImplementation ??= await CfeMacroImplementation.start(
+          protocol: Protocol(
+              encoding: ProtocolEncoding.json,
+              version: ProtocolVersion.macros1),
+          packageConfig: Uri.file(packageConfigPath));
+      injected_cfe.macroImplementation = cfeMacroImplementation;
+    } else {
+      injected_cfe.macroImplementation = null;
+    }
+
+    final fileResults = <FileResult>[];
+    final stopwatch = Stopwatch()..start();
+
+    // Don't directly use the compiler output: for consistency with the analyzer
+    // codepath, run from the resulting source.
+    // TODO(davidmorgan): maybe offer both as options? Not clear yet.
+    final outputFile = File('/dev/null');
+
+    final packagesUri = Uri.file(packageConfigPath);
+
+    final computeKernelResult = await computeKernel([
+      '--enable-experiment=macros',
+      '--no-summary',
+      '--no-summary-only',
+      '--target=vm',
+      '--dart-sdk-summary=${_productPlatformDill.uri}',
+      '--output=${outputFile.path}',
+      for (final sourceFile in sourceFiles) '--source=${Uri.file(sourceFile)}',
+      '--packages-file=$packagesUri',
+      // TODO(davidmorgan): this is so we can pull the generated
+      // augmentation source out of incremental compiler state; find a less
+      // hacky way.
+      '--use-incremental-compiler',
+      // For augmentations.
+      '--enable-experiment=macros',
```

Comment:
Is this intended to be `augmentations`?
```suggestion
      '--enable-experiment=augmentations',
```

Reply:
We might also need enhanced-parts